File: frontend/lib/types/bookmark.ts
/**
 * Bookmark types
 */

export interface Bookmark {
  id: string;
  user_id: string;
  question_id: string;
  notes: string | null;
  created_at: string;
  updated_at: string | null;
}

export interface BookmarkWithQuestion extends Bookmark {
  question_stem: string;
  question_status: string;
  year_id: number | null;
  block_id: number | null;
  theme_id: number | null;
  difficulty: string | null;
  cognitive_level: string | null;
}

export interface CreateBookmarkRequest {
  question_id: string;
  notes?: string | null;
}

export interface UpdateBookmarkRequest {
  notes?: string | null;
}

export interface CheckBookmarkResponse {
  is_bookmarked: boolean;
  bookmark_id: string | null;
}

File: backend/app/schemas/bookmark.py
"""Pydantic schemas for bookmarks."""

from datetime import datetime
from uuid import UUID

from pydantic import BaseModel, Field


# ============================================================================
# Bookmark Schemas
# ============================================================================


class BookmarkCreate(BaseModel):
    """Request to create a bookmark."""

    question_id: UUID = Field(..., description="Question ID to bookmark")
    notes: str | None = Field(None, description="Optional user notes")


class BookmarkUpdate(BaseModel):
    """Request to update a bookmark's notes."""

    notes: str | None = Field(None, description="Updated notes")


class BookmarkOut(BaseModel):
    """Bookmark response."""

    id: UUID
    user_id: UUID
    question_id: UUID
    notes: str | None
    created_at: datetime
    updated_at: datetime | None

    class Config:
        from_attributes = True


class BookmarkWithQuestion(BaseModel):
    """Bookmark with question details."""

    id: UUID
    user_id: UUID
    question_id: UUID
    notes: str | None
    created_at: datetime
    updated_at: datetime | None

    # Question details
    question_stem: str
    question_status: str
    year_id: int | None
    block_id: int | None
    theme_id: int | None
    difficulty: str | None
    cognitive_level: str | None
